web/controllers/pair_retro_controller.ex

```diff
@@ -13,28 +13,48 @@ defmodule Pairmotron.PairRetroController do
   end
 
   def new(conn, %{"pair_id" => pair_id}) do
-    conn = assign(conn, :projects, Repo.all(Project))
-    current_user = conn.assigns[:current_user]
-    changeset = PairRetro.changeset(%PairRetro{}, %{pair_id: pair_id, user_id: current_user.id}, nil)
-    render(conn, "new.html", changeset: changeset)
+    current_user = conn.assigns.current_user
+
+    pair = Pair.pair_with_users(pair_id) |> Repo.one
```

Comment:
So because this uses a query and `Repo.one` instead of `Repo.get |> Repo.preload`, not only is it only one database call, but it also elegantly handles when no record exists.

query `|> Repo.one` returns nil if no record is found, while `Repo.get |> Repo.preload` throws an exception if no record is found. This lets us handle it with the `is_nil(pair)` case very cleanly.